system/metrics/sinks/file/fileSink.cpp

```diff
@@ -65,12 +65,39 @@ void FileMetricSink::writeMeasurementToFile(const std::shared_ptr<IMetric> &pMet
     }
 
     const char *unitsStr = pManager->queryUnitsString(pMetric->queryUnits());
-    if (unitsStr)
+
+    if (pMetric->queryMetricType() != METRICS_HISTOGRAM)
     {
-        name.append(".").append(unitsStr);
+        if (unitsStr)
+        {
+            name.append(".").append(unitsStr);
+        }
+        fprintf(fhandle, "%s -> %" I64F "d, %s\n", name.c_str(), pMetric->queryValue(), pMetric->queryDescription().c_str());
     }
+    else
+    {
+        std::vector<__uint64> values = pMetric->queryHistogramValues();
+        std::vector<__uint64> limits = pMetric->queryHistogramBucketLimits();
+        size_t numValues = values.size();
```

Comment:
numValue is perfectly legit, but could be misunderstood as numeric values, etc. this could be refactored as valuesCount, etc.